clang/docs/checkedc/Update-to-latest-LLVM-sources.md

```diff
@@ -1,81 +1,212 @@
-
-The first step is to create updated baseline branches:
-1. Create new branches of your local baseline branches (we suggest creating new
-branches so that you can run automated testing. You'll need new branches
-that you can push to GitHub).
-2. Update those branches to the latest sources.
-3. Run testing on those branches to make sure things are stable.
-
-The second step is to create updated master branches:
-  1. Create branches of your updated baseline branches.
-  2. Merge changes from the Checked C master branches into those branches.
-  3. Fix merge conflicts and run testing.  You will likely need to fix some issues
-     and re-run testing until all issues are fixed.
-
-The third step is to merge your changes back into your baseline and master branches.
-
-## Create updated branches of your baseline branches
-
-First create remote to the mirrored GitHub repo that contains the updated sources
-for LLVM/Clang. Go to your LLVM/Clang repo and do:
-
-    git remote add mirror https://github.com/llvm-mirror/llvm-project
-
-Then branch your baseline branch and merge changes into it:
-
+We are staying in sync with the upstream LLVM/Clang repository. The `baseline`
+branch of the `checkedc-clang` repository is a pristine copy of LLVM/Clang
+sources, and the `master` branch contains LLVM/Clang sources plus the support
+for Checked C language extensions. We periodically upgrade the `baseline` branch
+and the `master` branch to sources of the most recent releases from the upstream
+LLVM/Clang repository.
+
+The branching and merging policies of the `checkedc-clang` repository are
+aligned with the versioning and branching of upstream LLVM/Clang repository in
+order to closely follow the LLVM/Clang releases. LLVM/Clang releases are
+versioned as &lt;major-version&gt;.&lt;minor-version&gt;.&lt;patch-version&gt;.
+The LLVM/Clang repository has a branch called `main`, and several release
+branches (created off `main`) called `release/<major-version>.x`, each
+corresponding to the major version number of a release. New features go on the
+`main` branch. At an appropriate commit on the `main` branch, a release branch
+is created. Bug fixes go on the release branch and only some of these bug fixes
+get merged back into the `main` branch. Typically, LLVM/Clang makes two final
+releases for each major version.
+
+Now we describe our approach for upgrading the `checkedc-clang` repository to
+LLVM/Clang sources corresponding to the two releases for each major version,
+for example, 12.0.0 and 12.0.1. The upgrade is performed in two phases as
+described below. This description is followed by detailed instructions for each
+phase.
+
+**Phase 1**: We upgrade the `master` branch of the `checkedc-clang` repository
+up to the commit hash on the `main` branch of the LLVM/Clang repository at which
+the branch `release/12.x` is created - we shall refer to this commit hash as
+&lt;branch-point-of-12-on-main&gt;. Note that &lt;branch-point-of-12-on-main&gt;
```

Comment:
Current LLVM practice seems to be to create a tag `llvmorg-13-init` referring to the commit _after_ the branch point for the LLVM 12 release, so you should be able to use `llvmorg-13-init^` as a commit expression referring to the branch point (`^` means "previous commit") rather than looking up the commit ID manually.

Reply:
We are not looking up the commit ID manually - we used the command `git merge-base main release/12.x` in a clone of the pristine LLVM/Clang repository. I have updated this in the document (I had missed it earlier). 

Additionally, 
   1) the commands `git merge-base main release/12.x` and `git show llvmorg-12-init^` show different commit hashes, and 
   2) we want &lt;branch-point-of-12-on-main&gt; to be a commit hash on the `main` branch.

Therefore, the document currently does not recommend the use of `llvmorg-12-init^` as &lt;branch-point-of-12-on-main&gt;. We may investigate this further during the next LLVM/Clang upgrade.

All other review comments have been incorporated.

Reply:
> 1. the commands `git merge-base main release/12.x` and `git show llvmorg-12-init^` show different commit hashes, and

That should be `git show llvmorg-13-init^`, which is indeed the same as `git merge-base main release/12.x`.  Still, `git merge-base main release/12.x` may be a clearer way to express what we want; good idea.

Reply:
Okay! Thanks for the clarification.